_posts/2024-08-23-Lumi_web_introduction.md

```diff
@@ -0,0 +1,222 @@
+
+### Programming Quantum Circuit
+
+Start by creating a Python file to set up your desired quantum circuit. To run your quantum circuit on Helmi, you need to connect to the Helmi backend using a specific URL (Helmi Cortex URL). This URL is pre-configured in the LUMI environment, so you just need to fetch it from the environment variables in your Python script. The example in `example.py` below demonstrates how to do this and set up your quantum circuit:
+
+```python
+import os
+from iqm.qiskit_iqm import IQMProvider
+from qiskit.compiler import transpile
+from qiskit import QuantumCircuit
+from qiskit.visualization import plot_histogram
+
+
+# Set up the Helmi backend
+## The Helmi Cortex URL is pre-configured in the LUMI environment, so you only need to fetch it from the OS to gain access to Helmi.
+
+HELMI_CORTEX_URL = os.getenv('HELMI_CORTEX_URL')
+if not HELMI_CORTEX_URL:
+    raise ValueError("Environment variable HELMI_CORTEX_URL is not set")
+
+provider = IQMProvider(HELMI_CORTEX_URL)
+backend = provider.get_backend()
+
+  
+# Create a bell state: 
+qc = QuantumCircuit(2)
+qc.h(0)
+qc.cx(0, 1)
+qc.measure_all() #expected equal majority counts of '00' and '11'
+
+print("My quantum circuit:")
+print(qc.draw())
+
+
+# Transpile the circuit to Helmi native gates, and map it to the star-shape topology backend
+transpiled_circuit = transpile(qc, backend, optimization_level=3)
+
+print("\nTranspiled quantum circuit:")
+print(transpiled_circuit.draw())
+  
+# Execute the transpiled circuit on the backend
+job = backend.run(transpiled_circuit, shots=1000)
+result = job.result()
+counts = result.get_counts()
+
+print("\nCounts result:")
+print(counts)
+
+# You can also plot the histogram and save it into your lumi workspace
+plot_histogram(counts).savefig("histogram.png")
+```
+
+### Sbatch script to run on LUMI
+
+To run your Python file on the LUMI system, you'll need to create an sbatch script. This script is necessary for submitting your job to the LUMI job scheduler, which allocates resources and manages the execution of your job.
+
+Create a `script.sh` file with the following content:
+
+```bash
+#!/bin/bash -l
+
+#SBATCH --job-name=example # Job name
+#SBATCH --output=./helmijob.o%j # Name of stdout output file followed by the job ID
+#SBATCH --error=./helmijob.e%j # Name of stderr error file followed by the job ID
+#SBATCH --partition=xxx # Partition (queue) name, replace xxx with correct partition
+#SBATCH --ntasks=1 # One task (process)
+#SBATCH --cpus-per-task=1 # Number of cores (threads)
+#SBATCH --time=00:15:00 # Run time (hh:mm:ss)
+#SBATCH --account=project_xxx # Project for billing, replace xxx with your own project_id
+#SBATCH --mem-per-cpu=1G # Memory per CPU
+
+
+module use /appl/local/quantum/modulefiles
+module load helmi_qiskit # Load the module to use qiskit on Helmi
+
+# module load helmi_cirq # Load the module to use cirq on Helmi
+
+# Save the job ID to a file for later reference
+echo $SLURM_JOB_ID >> job_id.txt
```

Comment:
Not needed I would say. It's more useful to save the cortex job ID from Helmi. See my other suggestion.